Docker Desktop 4.31.0 (#20114)
docker/docker.github.io: content/desktop/release-notes.md

- Fixed a bug where Docker Debug was not working properly with Enhanced Container Isolation enabled.
- Fixed a bug where UDP responses were not truncated properly.
- Fixed a bug where the Update screen was hidden when using [Settings Management](hardened-desktop/settings-management/_index.md).
- Fixed [docker/for-mac#7274](https://github.com/docker/for-mac/issues/7274)

#### For Windows

- Changed the `--allowed-org` installer flag to write a policy registry key instead of to the `registry.json`.

#### For Mac

- Moved the setting **Automatically check configuration** from **Advanced** settings to **General** settings.
- Improved VirtioFS caching by implementing longer attributes timeout and invalidation.

#### For Linux

- Added Linux headers to the VM, to ease the compilation of custom kernel modules.

### Security

#### For all platforms

- Fixed a security bug in Enhanced Container Isolation (ECI) mode where a user could create Docker volumes sourced from restricted directories inside the Docker Desktop VM and mount them into containers, thereby giving the container access to such restricted VM directories.

### Deprecation

#### For Windows

> - The CLI binary that used to be shipped as `com.docker.cli` is now shipped simply as `docker`. This release leaves the CLI binary as `com.docker.cli`, but it will be removed next release.

Author: Indeed!

Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/projects/1) to see what's coming next.

## 4.31.0

{{< release-date date="2024-05-30" >}}

{{< desktop-install all=true beta_win_arm=true version="4.31.0" build_path="/TODO_NOT_AVAILABLE_YET/" >}}

### New

- Admins can now use alternative methods to the `registry.json` file, to enforce sign in.
- [Air-Gapped Containers](desktop/hardened-desktop/air-gapped-containers.md) is now generally available. 
- Introducing: compose bridge. You might be asked your password to install it. Make sure in settings->advanced->Automatically check configuration is turned on.

> - Added Multi-node Kind as a Kubernetes option (Beta).

Reviewer: ```suggestion
```
Remove this line as well. Unfortunately, @milas's departure from our team has delayed our multi-node Kind release for DD. We're still scoping the updated timeline given impacts to our team velocity.

> - Added a Docker Compose documentation helper and a new way to view Compose files with the Compose File Viewer (Beta).

Reviewer: Suggesting an edit for better describing this feature:

```suggestion
- Docker Compose File Viewer shows your Compose YAML with syntax highlighting and contextual links to relevant docs (Beta).
```

- [Air-Gapped Containers](desktop/hardened-desktop/air-gapped-containers.md) is now generally available. 
- Added Multi-node Kind as a Kubernetes option (Beta).

#### For all platforms

- Introducing: compose bridge. You might be asked your password to install it. Make sure in settings->advanced->Automatically check configuration is turned on.
- Added a Docker Compose documentation helper and a new way to view Compose files with the Compose File Viewer (Beta).

### Upgrades

- [Docker Engine and CLI v26.1.3](https://github.com/moby/moby/releases/tag/v26.1.3).
- [Docker Scout CLI v1.9.1](https://github.com/docker/scout-cli/releases/tag/v1.9.1)
- [Docker Compose v2.27.1](https://github.com/docker/compose/releases/tag/v2.27.1)
- [Docker Buildx v0.14.1](https://github.com/docker/buildx/releases/tag/v0.14.1)
- [Containerd v1.6.32](https://github.com/containerd/containerd/releases/tag/v1.6.32)
- [Credential Helpers v0.8.2](https://github.com/docker/docker-credential-helpers/releases/tag/v0.8.2)
- Linux kernel `v6.6.31`

### Bug fixes and enhancements

#### For all platforms

- Allow newer release to be displayed when an update has been already downloaded.

> - Moved the setting **Automatically check configuration** from **Advanced** settings to **General** settings.

Author: Really? okok

- Fixed a bug where Docker Debug was not working properly with Enhanced Container Isolation enabled.
- Fixed a bug where UDP responses were not truncated properly.
- Fixed a bug where the Update screen was  hidden when using [Settings Management](hardened-desktop/settings-management/_index.md).

> - Fixes [docker/for-mac#7274](https://github.com/docker/for-mac/issues/7274)

Author: I don't think so looking at the issue and the description

#### For Windows

- Changed the `--allowed-org` installer flag to write a policy registry key instead of to the `registry.json`.

#### For Mac

- Improved VirtioFS caching by implementing longer attributes timeout and invalidation.

#### For Linux

- Added Linux headers to the VM, to ease the compilation of custom kernel modules.

### Security

#### For all platforms

- Fixed a security bug in Enhanced Container Isolation (ECI) mode where a user could create Docker volumes sourced from restricted directories inside the Docker Desktop VM and mount them into containers, thereby giving the container access to such restricted VM directories.

> ### Deprecation

Reviewer 2: cc @laurazard @krissetto - we should probably mention the removal of the "cloud integration" wrapper (and the "com.docker.cli" to be removed in the next release)

Reviewer 3: ```suggestion
### Deprecation
- The CLI binary that used to be shipped as `com.docker.cli` is now shipped simply as `docker`. This release leaves the CLI binary as `com.docker.cli`, but it will be removed next release.
```

Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/projects/1) to see what's coming next.

## 4.31.0

{{< release-date date="2024-05-30" >}}

{{< desktop-install all=true beta_win_arm=true version="4.31.0" build_path="/TODO_NOT_AVAILABLE_YET/" >}}

### New

> 

Reviewer 4: ```suggestion
- New Sidebar user experience
```

I'm not sure if this is where it's supposed to go.

- Newer releases are now displayed when an update has already been downloaded.
- Added `proxyEnableKerberosNTLM` config to `settings.json` to enable fallback to Basic proxy authentication if Kerberos/NTLM environment is not properly set up.
- Fixed a bug where Docker Debug was not working properly with Enhanced Container Isolation enabled.
- Fixed a bug where UDP responses were not truncated properly.
- Fixed a bug where the Update screen was hidden when using [Settings Management](hardened-desktop/settings-management/_index.md).
- Fixed a bug where proxy settings defined in `admin-settings.json` were not applied correctly on startup.
- Fixed [docker/for-mac#7274](https://github.com/docker/for-mac/issues/7274)

#### For Windows

- Changed the `--allowed-org` installer flag to write a policy registry key instead of to the `registry.json`.

#### For Mac

- Moved the setting **Automatically check configuration** from **Advanced** settings to **General** settings.
- Improved VirtioFS caching by implementing longer attributes timeout and invalidation.

#### For Linux

- Added Linux headers to the VM, to ease the compilation of custom kernel modules.

### Security

> #### For all platforms

Reviewer 5: Probably worth mentioning, I suppose this would be a good place (cc @doringeman) : 

By default, only extensions listed in the marketplace can be installed in Docker Desktop. This can be changed in Docker Desktop settings. Extension developers will need to change this option in order to test their extensions.

(happy to change the wording :) )

Author: Added!

Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/projects/1) to see what's coming next.

## 4.31.0

{{< release-date date="2024-05-30" >}}

{{< desktop-install all=true beta_win_arm=true version="4.31.0" build_path="/TODO_NOT_AVAILABLE_YET/" >}}

### New

> - Administrators can now use alternative methods to the `registry.json` file to enforce sign in.

Reviewer 6: Can we include the link to the "Enforce sign in" page?

- [Air-Gapped Containers](desktop/hardened-desktop/air-gapped-containers.md) is now generally available. 
- Docker Compose File Viewer shows your Compose YAML with syntax highlighting and contextual links to relevant docs (Beta, progressive rollout).
- New Sidebar user experience.

### Upgrades

- [Docker Engine and CLI v26.1.3](https://github.com/moby/moby/releases/tag/v26.1.3).
- [Docker Scout CLI v1.9.1](https://github.com/docker/scout-cli/releases/tag/v1.9.1)
- [Docker Compose v2.27.1](https://github.com/docker/compose/releases/tag/v2.27.1)
- [Docker Buildx v0.14.1](https://github.com/docker/buildx/releases/tag/v0.14.1)
- [Containerd v1.6.32](https://github.com/containerd/containerd/releases/tag/v1.6.32)
- [Credential Helpers v0.8.2](https://github.com/docker/docker-credential-helpers/releases/tag/v0.8.2)
- Linux kernel `v6.6.31`

### Bug fixes and enhancements

#### For all platforms

> - Newer releases are now displayed when an update has already been downloaded.

Reviewer 6: ```suggestion
- Newer releases are now displayed in the **Software updates** when an update has already been downloaded.
```

- Added `proxyEnableKerberosNTLM` config to `settings.json` to enable fallback to Basic proxy authentication if Kerberos/NTLM environment is not properly set up.
- Fixed a bug where Docker Debug was not working properly with Enhanced Container Isolation enabled.
- Fixed a bug where UDP responses were not truncated properly.
- Fixed a bug where the Update screen was hidden when using [Settings Management](hardened-desktop/settings-management/_index.md).
- Fixed a bug where proxy settings defined in `admin-settings.json` were not applied correctly on startup.

> - Fixed [docker/for-mac#7274](https://github.com/docker/for-mac/issues/7274)

Reviewer 6: ```suggestion
- Fixed a bind mount bug where a file modified on host was not updated in the container, when gRPC FUSE file sharing is used on macOS and on Windows with Hyper-V. Fixes [docker/for-mac#7274](https://github.com/docker/for-mac/issues/7274), [docker/for-win#14060](https://github.com/docker/for-win/issues/14060).
```

Reviewer 6: @fredericdalleau please review the wording

Author: Possibly `- Fixed a bug where a bind mounted file modified on host was not updated in the container, when gRPC FUSE file sharing is used on macOS and on Windows with Hyper-V. Fixes [docker/for-mac#7274](https://github.com/docker/for-mac/issues/7274), [docker/for-win#14060](https://github.com/docker/for-win/issues/14060).` for consistency?

Author: (all bug entry here starts with `fixed a bug where [...]`

Reviewer 7: It's actually a partial fix.

Reviewer 7: - I'd suggest `Fixed a bug where a bind mounted file modified on host is not updated after the container restarts, when gRPC FUSE file sharing is used on macOS and on Windows with Hyper-V.`

Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/projects/1) to see what's coming next.

## 4.31.0

{{< release-date date="2024-05-30" >}}

{{< desktop-install all=true beta_win_arm=true version="4.31.0" build_path="/TODO_NOT_AVAILABLE_YET/" >}}

### New

- Tips for existing Compose files with Compose File Viewer.
- Enable alternatives to Registry.JSON for enforcing sign in.

> - Air-gapped containers.

Author: Is the page not created yet?

> - Air-gapped containers.

Author: Found it, it needs to be `desktop/hardened-desktop/air-gapped-containers.md` instead 😁
